src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, hidden) {
+    super(name, hidden);
+    this.hidden = false;
+  }
+
+  hide() {
+    if (!this.hidden) {
+      this.hidden = true;
+    } else {
+      this.hidden = false;
+    }
```

Comment:
Let's simplify this with a toggler function.
```suggestion
    this.hidden = !this.hidden;
```